src/openfermion/utils/_trotter_qasm_conversion.py

```diff
@@ -0,0 +1,331 @@
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		third_order_trotter_helper(trotterized_list, atemp, btemp)
+
+
+
+
+def trotterize(
+	hamiltonian,trotter_steps = 1, 
+	trotter_order = 1, 
+	term_ordering = [], 
+	k_exp = 1.):
+	'''
+	This function trotterizes a hamiltonian to an order 1-3 and 
+	for a given number of trotter steps
+
+	Params:
+		hamiltonian: full hamiltonian as a QubitOperator.
+		trotter_steps: number of trotter steps as an integer.
+		order: order of trotterization as an integer from 1-3.
+		term_ordering (optional): list of tuples (QubitOperator terms
+				dictionary keys) that specifies order of terms when
+				trotterizing.
+		k_exp (optional): exponential factor to all terms when trotterizing.
+	
+	Returns:
+		tuple: (A list of single Pauli-term QubitOperators, the number of 
+		qubits required by the hamiltonian as an integer)
+	
+	Raises:
+		ValueError if order > 3 or order <= 0 or trotter_steps > 100,
+			TypeError for incorrect type 
+	'''
+	
+
+	if trotter_order > 3 or trotter_order <= 0:
+		raise ValueError("Invalid trotter order: " + str(order))
+	elif trotter_steps <= 0 or trotter_steps > 100: 
+		raise ValueError("Invalid number of trotter steps: " + str(trotter_steps))
+		
+	if not isinstance(hamiltonian, QubitOperator):
+		raise TypeError("Hamiltonian must be a QubitOperator.")
+	if not hamiltonian.terms:
+		raise TypeError("Hamiltonian must be a non-empty QubitOperator.")
+	if not isinstance(trotter_steps, int):
+		raise TypeError("trotter_steps must be an int")
+	if not isinstance(trotter_order, int):
+		raise TypeError("trotter_order must be an int")
+	if not isinstance(k_exp, float):
+		raise TypeError("k_exp must be a float")
+		
+		
+	num_qubits = 0
+	for term in hamiltonian.terms.items():
+		for item in term[0]:
+			if item[0] > num_qubits:
+				num_qubits = item[0]
+	num_qubits += 1
+	ret_val = []
+	
+	if len(term_ordering) == 0:
+		for term in hamiltonian.terms.items():
+			term_ordering.append(term[0])
+	
+	if trotter_order == 1:
+		for step in range(trotter_steps):
+			for op in term_ordering:
+				ret_val.append(QubitOperator(op,hamiltonian.terms[op]) 
+					* k_exp/trotter_steps)
+
+	elif trotter_order == 2:
+		if len(term_ordering) < 2:
+			raise ValueError("Not enough terms in the Hamiltonian to do "+\
+				"second order trotterization")
+		for op in term_ordering[:-1]:
+			ret_val.append(QubitOperator(op,hamiltonian.terms[op]) * k_exp 
+				/ (2.0 * trotter_steps))
+		ret_val.append(QubitOperator(term_ordering[-1],
+			hamiltonian.terms[term_ordering[-1]] * k_exp / trotter_steps))
+
+		for op in reversed(term_ordering[:-1]):
+			ret_val.append(QubitOperator(op,hamiltonian.terms[op]) * k_exp 
+				/ (2.0 * trotter_steps))
+
+		for step in range(trotter_steps - 1):
+			ret_val.extend(ret_val)
+
+	elif trotter_order == 3:
+		if len(term_ordering) < 2:
+			raise ValueError("Not enough terms in the Hamiltonian to do "+\
+				"third order trotterization")
+		hamiltonian /= trotter_steps
```

Comment:
The coefficients are probably all floats... but just in case maybe use /= float(trotter_steps) to avoid integer divsion in python2.  This would have no effect in python3